pkg/controllers/dnspolicy/dnspolicy_controller.go

```diff
@@ -195,8 +195,7 @@ func (r *DNSPolicyReconciler) deleteResources(ctx context.Context, dnsPolicy *v1
 	if err != nil {
 		return err
 	}
-
-	if err := r.reconcileDNSRecords(ctx, dnsPolicy, gatewayDiffObj); err != nil {
+	if err = r.deleteDNSRecords(ctx, dnsPolicy); err != nil {
```

Comment:
:+1: 

I think you  will need to do the same for `reconcileHealthChecks` though.